### PR TITLE
edit docs for 0.3 release

### PR DESCRIPTION
**Pull request by Bradamant3**
Mostly classic writer nits to follow up on #159, a few additional fixes:
- removed "Known limitations" from readme since this release addresses both of them
- added a bit more navigation

Signed-off-by: JENNIFER RONDEAU <jrondeau@heptio.com>

### CODE REVIEW

File: docs/tls.md [hunk 1]
```diff
@@ -1,22 +1,22 @@
-# TLS Support
+# TLS support
```
Comment (davecheney) [inline docs/tls.md]: I'm delighted you're not a fan of Title Case

Comment (Bradamant3) [reply docs/tls.md]: :D We aim to Please

Comment (Bradamant3) [reply docs/tls.md]: In other matters -- still need to push edits to proxy-proto.md. Doesn't tls.md provide the same content?

Comment (davecheney) [reply docs/tls.md]: yes, there is probably overlap between tls.md and proxy-proto.md. My vote would be to push the complexity into the latter -- not everyone uses aws.

**Pull request merged**